src/features/codeLensProvider.ts

```diff
@@ -52,20 +51,19 @@ export default class OmniSharpCodeLensProvider extends AbstractProvider implemen
         'ToString': true
     };
 
-    provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken): vscode.CodeLens[] | Thenable<vscode.CodeLens[]> {
-        if (!this._options.showReferencesCodeLens && !this._options.showTestsCodeLens)
-        {
+    async provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken) {
+        if (!this._options.showReferencesCodeLens && !this._options.showTestsCodeLens) {
             return [];
         }
 
-        return serverUtils.currentFileMembersAsTree(this._server, { FileName: document.fileName }, token).then(tree => {
-            let ret: vscode.CodeLens[] = [];
-            tree.TopLevelTypeDefinitions.forEach(node => this._convertQuickFix(ret, document.fileName, node));
-            return ret;
-        });
+        let tree = await serverUtils.currentFileMembersAsTree(this._server, { FileName: document.fileName }, token);
+        let ret: vscode.CodeLens[] = [];
+        tree.TopLevelTypeDefinitions.forEach(node => this._convertQuickFix(ret, document.fileName, node));
```

Comment:
Does there need to be some sort of await here for `_convertQuickFix`? Or does `forEach` handle awaiting each promise before returning?